docs/reference/api/connectivity/bluetooth/GattServer.md

```diff
@@ -1,6 +1,31 @@
 ## GattServer
 
-[Add description here.]
+A GattServer is a collection of GattServices. These services contain characteristics that a peer connected to the device may read or write. These characteristics may also emit updates to subscribed clients when their values change.
+
+#### Server layout
+
+Application code can add a GattService object to the server with the help of the function `addService()`. That function registers all the GattCharacteristics enclosed in the service, as well as all the characteristic descriptors (see GattAttribute) that these characteristics contain. Service registration assigns a unique handle to the various attributes that are part of the service, and this handle should be used to subsequently read or write these components.
+
+There are no defined primitives that remove a single service; however, a call to the function `reset()` removes all services previously registered in the GattServer.
+
+#### Characteristic and attributes access
+
+Values of the characteristic and the characteristic descriptor present in the GattServer must be accessed through the handle assigned to them when the service has been registered. The GattServer class offers several flavours of `read()` and `write()` functions that retrieve or mutate an attribute value.
+
+If a client has subscribed to a given characteristic's value update, then the application code can query by invoking the function `areUpdatesEnabled()`.
+
+#### Events
+
+The GattServer allows application code to register several event handlers that can be used to monitor client and server activities:
+
+- `onDataSent`: Register an event handler that is called when a characteristic value update has been sent to a client.
+- `onDataWriten`: Register an event handler that is called when a client has written an attribute of the server.
+- `onDataRead`: Register an event handler that is called when a client has read an attribute of the server.
+- `onUpdatesEnabled`: Register an event handler that is called when a client subscribes to updates for a characteristic.
+- `onUpdatesDisabled`: Register an event handler that is called when a client unsubscribes from updates for a characteristic.
+- `onConfimationReceived`: Register an event handler that is called when a client acknowledges a characteristic value notification.
+
+The term characteristic value update is used to represent Characteristic Value Notification and Characteristic Value Indication when the nature of the server initiated is not relevant.
```

Comment:
Query: Could we say "The term "characteristic value update" represents Characteristic..."?